includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -54,4 +54,29 @@ public function add_or_append_attribute( &$attributes, $key, $value, $separator
 			$attributes[ $key ] = $value;
 		}
 	}
+
+	/**
+	 * Decide if we should remove a src attribute if https is required.
+	 * If not required, the implementing class may want to try and force https instead.
+	 *
+	 * @param string $src
+	 * @param boolean $force_https
+	 * @return string
+	 */
+	public function validate_src_attribute( $src, $force_https = false ) {
```

Comment:
This might be better named `maybe_enforce_https_src` or something like that, since we're not actually validating anything.